threescale/src/utils.rs

```diff
@@ -75,3 +83,36 @@ pub fn add_hierarchy_to_metrics(hierarchy: &Hierarchy, metrics: &mut Metrics) {
         }
     }
 }
+
+/// AllocatedSize impl for String.
+impl AllocatedSize for String {
+    // Returns the dynamically allocated value of a String on the heap.
+    fn dynamic_allocated_size(&self) -> usize {
+        self.capacity()
+    }
+}
+
+/// AllocatedSize impl for AppIdentifier.
+impl AllocatedSize for AppIdentifier {
+    // Returns the dynamically allocated value of a AppIdentifier on the heap.
+    fn dynamic_allocated_size(&self) -> usize {
+        match self {
+            AppIdentifier::AppId(app_id, None) => app_id.as_ref().to_string().capacity(),
+            AppIdentifier::AppId(app_id, Some(app_key)) => {
+                app_id.as_ref().to_string().capacity() + app_key.as_ref().to_string().capacity()
+            }
+            AppIdentifier::UserKey(user_key) => user_key.as_ref().to_string().capacity(),
+        }
```

Comment:
Since AppId and UserKey are structs wrapping `String`, can we get **its** capacity? Here we are converting `&str` to `String` and getting its capacity. If not, how about just counting the number of characters and we know each char = 1 byte.

Reply:
I am not sure if it will work but implementing AllocatedSize for AppId and UserKey is worth a try.

Reply:
Looks like @unleashed mentioned the same things above. So, you can ignore my comments.

Reply:
@rahulanand16nov This is no longer required as suggested by @unleashed . Dynamic allocations are not considered.